src/kw_ssh.sh

```diff
@@ -0,0 +1,127 @@
+#   (--command|-c) and (--script|-s). If this parameter receives a null value
+#   this function will perform a simple ssh connection; otherwise, it will
+#   attempt to execute a command or script on the remote host.
+function kw_ssh()
+{
+  local port
+  local script_path
+  local cmd
+  local ssh_cmd
+  local flag
+
+  parser_ssh_options "$@"
+  if [[ "$?" -gt 0 ]]; then
+    complain "Invalid option: ${options_values['ERROR']}"
+    exit 22 # EINVAL
+  fi
+
+  user="${remote_parameters['REMOTE_USER']}"
+  remote="${remote_parameters['REMOTE_IP']}"
+  port="${remote_parameters['REMOTE_PORT']}"
+  script_path=${options_values['SCRIPT']}
+  cmd=${options_values['CMD']}
+  flag=${options_values['TEST_MODE']}
+
+  if [[ -n "${options_values['SCRIPT']}" ]]; then
+    if [[ ! -f "$script_path" ]]; then
+      complain "No such file: \"$script_path\""
+      exit 2 # ENOENT
+    fi
+
+    ssh_cmd="\"bash -s\" -- < $script_path"
+  elif [[ -n "${options_values['CMD']}" ]]; then
+    ssh_cmd="${options_values['CMD']}"
+  fi
+
+  # Add port
+  if [ -n "$port" ]; then
+    port="-p $port"
+  fi
+
+  cmd_manager "$flag" "ssh $port $user@$remote $ssh_cmd"
+}
+
+function parser_ssh_options()
+{
+  local options
+  local short_options
+  local long_options
+  local transition_variables
+
+  long_options='help,test_mode,script:,command:'
+  short_options='h,s:,c:'
+
+  options="$(getopt -q --options "$short_options" \
+    --longoptions "$long_options" \
+    -- "$@")"
+
+  if [[ "$?" != 0 ]]; then
+    options_values['ERROR']="$(getopt --name 'ssh' --options "$short_options" \
+      --longoptions "$long_options" \
+      -- "$@" 2>&1 > /dev/null)" # Get errors from stderr
+    return 22                    # EINVAL
+  fi
+
+  options_values['SCRIPT']=''
+  options_values['COMMAND']=''
+  options_values['TEST_MODE']=''
+
+  # Set default values
+  if [[ -n ${configurations[default_deploy_target]} ]]; then
+    transition_variables=${configurations[default_deploy_target]}
+    options_values['TARGET']=${deploy_target_opt[$transition_variables]}
+  fi
+
+  populate_remote_info ''
+  if [[ "$?" == 22 ]]; then
+    options_values['ERROR']='Something is wrong in the remote option'
+    return 22 # EINVAL
+  fi
+
+  eval "set -- $options"
+  while [[ "$#" -gt 0 ]]; do
+    case "$1" in
+      --script | -s)
+        options_values['SCRIPT']+="$2"
+        shift 2
+        ;;
+      --command | -c)
+        options_values['CMD']+="$2"
+        shift 2
+        ;;
+      --help | -h)
+        ssh_help
+        exit
+        ;;
+      test_mode)
+        options_values['TEST_MODE']='TEST_MODE'
+        shift
+        ;;
+      --) # End of options, beginning of arguments
+        shift
+        ;;
+      *)
+        options_values['ERROR']="$1"
+        return 22
+        ;;
+    esac
+  done
+}
+
+function ssh_help()
+{
+  echo -e "kw ssh|s options:\n" \
+    "\tssh|s [--script|-s=\"SCRIPT PATH\"]\n" \
+    "\tssh|s [--command|-c=\"COMMAND\"]"
+}
```

Comment:
Why did you use the old help function? Shouldn't this use the new format with man support?